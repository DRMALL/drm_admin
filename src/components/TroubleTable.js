

import React from 'react'
import Table from 'antd/lib/table'
import { browserHistory } from 'react-router'
import TableTextHidden from './TableTextHidden'
import TableTime from './TableTime'
import Modal from 'antd/lib/modal'
import troubleDelById from '../actions/troubleDelById'
import dispatch from '../actions/dispatch'
import getRichEditorValue from '../utils/getRichEditorValue'
import delIcon from '../images/dels.png'

export default props => {
  const { troubleArr, troubleModal } = props.state.trouble
  const columns = [{
  title: '标题',
  dataIndex: 'title',
  key: 'title',
  width: '20%',
  render: text => TableTextHidden( text, 40 )
  },{
  title: '分类',
  dataIndex: 'category',
  key: 'category',
  width: '10%',
  render: text =>
  <div>{ text ? text.text : '' }</div>
  },{
  title: '内容',
  dataIndex: 'content',
  key: 'content',
  width: '30%',
  render: text => TableTextHidden( getRichEditorValue(text), 40 )
  },{
  title: '添加时间',
  dataIndex: 'createdAtTime',
  key: 'createdAtTime',
  width: '10%',
  render: text =>  TableTime( text, 15 )
  },{
  title: '操作',
  key: 'action',
  width: '10%',
  render: (text, record) =>
    <div className='troubel-cation' >
      <div onClick={ getTrouble } data-trouble-id={ record._id } >
        <i className="iconfont icon-compile trouble-table-icon"  style={{ color: '#0068d2', fontSize: 25 }} ></i>
      </div>
      <div className='user-arr-item' onClick={ delTroubel } data-trouble-id={ record._id } style={{ marginLeft: 10 }} >
          <img src={ delIcon } alt='删除' style={{ width:20,height:20, marginBottom:5 }} />
        </div>
    </div>
  },]
  return (
    <div>
      <div style={{paddingLeft: '20px', paddingRight: '20px', paddingBottom: '20px'}}>
        <Table dataSource={troubleArr} columns={columns} rowKey={ record => record._id } />
      </div>
      <Modal title='提示' visible={ troubleModal } okText='确定' cancelText='取消'
             onOk={ troubleDelById } onCancel={ cancleDeltrouble } >
             <p>确定删除此信息？</p>
      </Modal>
    </div>
    )
}

function delTroubel(e){
  const { troubleId } = e.currentTarget.dataset
  dispatch('TROUBLE_GET_TROUBLE_ID', troubleId )
}

function cancleDeltrouble(){
  dispatch('TROUBEL_CANCLE_DEL_TROUBEL')
}


function getTrouble(e){
  const { troubleId } = e.currentTarget.dataset
  localStorage.setItem('troubleId', troubleId )
  browserHistory.push('/trouble/edit')
  console.log(troubleId)
}
