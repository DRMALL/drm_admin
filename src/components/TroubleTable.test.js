

import TroubleTable from './TroubleTable'
import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

it('TroubleTable test', () => {
  const state = { trouble: { troubleArr: [{_id: 'sd345', title: 'z标题', content: 'sdaw34325235', category: { text: 'sfaer' } },{_id: 'sd3ds45', title: 'z标题', content: 'sdaw34325235',category: { text: 'sfaedsdr' } }] } }
  const wrapper = render(<TroubleTable state={ state } />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
