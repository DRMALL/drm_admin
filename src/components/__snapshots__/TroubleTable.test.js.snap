// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TroubleTable test 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-large ant-table-scroll-position-left"
        >
          <div
            class="ant-table-content"
          >
            <div
              class="ant-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col
                    style="width:20%;min-width:20%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:30%;min-width:30%;"
                  />
                  <col
                    style="width:20%;min-width:20%;"
                  />
                  <col
                    style="width:15%;min-width:15%;"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class=""
                    >
                      <span>
                        标题
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        分类
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        内容
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        添加时间
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        操作
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      <div>
                        z标题
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        sfaer
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        sdaw34325235
                      </div>
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        class="troubel-cation"
                      >
                        <div
                          data-trouble-id="sd345"
                        >
                          <i
                            class="iconfont icon-compile trouble-table-icon"
                            style="color:#0068d2;"
                          />
                        </div>
                        <div
                          class="user-arr-item"
                          data-trouble-id="sd345"
                        >
                          <i
                            class="iconfont icon-shanchu message-del-icon"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      <div>
                        z标题
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        sfaedsdr
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        sdaw34325235
                      </div>
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        class="troubel-cation"
                      >
                        <div
                          data-trouble-id="sd3ds45"
                        >
                          <i
                            class="iconfont icon-compile trouble-table-icon"
                            style="color:#0068d2;"
                          />
                        </div>
                        <div
                          class="user-arr-item"
                          data-trouble-id="sd3ds45"
                        >
                          <i
                            class="iconfont icon-shanchu message-del-icon"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-prev"
            tabindex="0"
            title="上一页"
          >
            <a />
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-next"
            tabindex="0"
            title="下一页"
          >
            <a />
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
